objects/src/notes/mod.rs

```diff
@@ -64,39 +76,40 @@ impl Note {
 
     /// Returns a commitment to this note.
     ///
-    /// The note hash is computed as hash(hash(hash(serial_num, [0; 4]), script_hash), vault_hash).
+    /// The note hash is computed as:
+    ///   hash(hash(hash(hash(serial_num, [0; 4]), script_hash), vault_hash), script_inputs_hash).
```

Comment:
I'd compute this different to satisfy the property that computing note hash can be done knowing only the `recipient` and `vault hash`. Specifically, we could probably do something like this:
```
hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)
```
Here, we define `recipient` as:
```
hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)
```

Reply:
What is the significance of `recipient` in this context?  Will this be used to populate the note prefix as has been mentioned in discussions via the grinding method?  Also what is the significant of not needing to know the `serial_number` and only needing to know its hash?

Reply:
For recipient: let's assume I'd like you to send me some asset `x`. To do this, you'll need to create a note which I can claim and which will contain asset `x`. To create this note, you'll need to execute a transaction which will output the hash of the created note.

If I give you just the hash of the note, it won't be enough for you to execute the transaction because you need to prove that the sum of input assets is equal to the sum of output assets. So, you need to know `x` and you need to know how to combine this `x` with something to get the note hash. This something is the `recipient`. If you know `x` and `recipient`, you can compute note hash as `hash(recipient, vault_root)`.

Of course, I could have given you all components of the note, and then you'd be able to compute the note hash as well, but I may not want to reveal anything else about the note to you. Giving the hash of all components except for the vault root, allows me to keep all other components private.

"Hiding" serial number behind the hash follows the same idea. If I give you a hash of serial number, you can create a note addressed to me, but you would know when/if I claimed the note because you won't know how to compute its nullifier (knowing hash of serial number does not enable you to compute the nullifier). I don't really have a strong use case of how this could be useful. So, if we can't think of something, we could simplify recipient computation as:
```
recipient = hash(hash(script_hash, input_hash), serial_num)
```

Reply:
makes sense! thanks for the explanation.